exe/shell/main.go

```diff
@@ -218,88 +269,123 @@ func addressCmd() (out string, err error) {
 	return
 }
 
-func threadCmd(args []string) (out string, err error) {
-	if len(args) == 0 {
-		err = fmt.Errorf("enter a thread name to use")
+func threadsCmd() (out string, err error) {
+	q, err := ds.Query(query.Query{Prefix: "/names"})
+	if err != nil {
+		return
+	}
+	all, err := q.Rest()
```

Comment:
Since you're just iterating through here anyway, maybe just use the iterator (`.Next()`)?

Reply:
ah yeah, that's better, though it's nice to have `i` to then deal with the newlines in the same loop - just gonna leave this for now - surely will change.